homeassistant/components/androidtv/media_player.py

```diff
@@ -491,6 +492,14 @@ def unique_id(self):
         """Return the device unique id."""
         return self._unique_id
 
+    async def _async_stop(self, event):
+        """Close the ADB socket connection."""
+        await self.aftv.adb_close()
+
+    async def async_added_to_hass(self):
+        """Run when entity is added to HA."""
+        self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self._async_stop)
```

Comment:
Why should we wait until the entity is added to home assistant to register the aftv adb close callback? The callback isn't related to the entity and the connection is opened before the entity is added to home assistant, right?

I'd expect to register the callback when opening the connection, before creating the entity.

Reply:
https://github.com/home-assistant/core/pull/37973/commits/a203dad5639743a4c88ff4633c449938ea2ba6b7

I moved it to right after creating the `AndroidTVDevice` / `FireTVDevice` instance.